src/test/java/org/broadinstitute/hellbender/tools/walkers/coverage/DepthOfCoverageIntegrationTest.java

```diff
@@ -0,0 +1,259 @@
+            return "";
+        }
+        Assert.fail("There was a problem loading DoCExtension: "+basename);
+        return null;
+    }
+
+    private File getExpectedDataDir() {
+        return getTestFile( "expected/");
+    }
+
+    @Test
+    public void testBaseOutputNoFiltering() throws IOException {
+        final String expectedBaseName = "depthofcoveragenofiltering";
+        final File baseOutputFile = createTempDir("depthofcoveragenofiltering");
+        final File output = IOUtils.createTempFileInDirectory( "depthofcoveragenofiltering", ".csv", baseOutputFile);
+
+        String cmd = "-R "+hg38Reference+" " +
+                "-I "+largeFileTestDir + "multiSampleSubsetted.bam " +
+                "-L "+ getTestFile("artificial.target_region.interval_list ")+
+                "--min-base-quality 0 -dels --print-base-counts -pt readgroup -pt sample -pt platform -pt library --output-format CSV -ct 10 -ct 15 -ct 20 -ct 25";
+        cmd += " -O "+output.getAbsolutePath();
+        runCommandLine(cmd.split(" "));
+
+        File[] actualFiles = baseOutputFile.listFiles();
+
+        compareOutputDirectories(expectedBaseName, output.getName(), actualFiles);
+    }
+
+    @Test
+    // Asserting that the per-base statistics are reasonable when run without an interval list file
+    public void testCoverageBehaviorWhenProvidedNoIntervalFile() throws IOException {
+        final String expectedBaseName = "testCoverageBehaviorWhenProvidedNoIntervalFile";
+        final File baseOutputFile = createTempDir("testCoverageBehaviorWhenProvidedNoIntervalFile");
+        final File output = IOUtils.createTempFileInDirectory( "testNoCoverageDueToFiltering", ".csv", baseOutputFile);
+
+        String cmd = "-R "+hg38Reference+" " +
+        "-I "+largeFileTestDir+"multiSampleSubsetted.bam"+
+        " -O "+output.getAbsolutePath() + " -L chr1:1570000-1571000 --omit-interval-statistics --omit-locus-table --omit-per-sample-statistics";
+        runCommandLine(cmd.split(" "));
+        File[] actualFiles = baseOutputFile.listFiles();
+
+        compareOutputDirectories(expectedBaseName, output.getName(), actualFiles);
+    }
+
+    @Test
+    // Coverage drops to nothing because '--max-base-quality 4' is set, resuling in very poor output coverage
+    public void testNoCoverageDueToFiltering() throws IOException {
+        final String expectedBaseName = "depthofcoveragenwithiltering";
+        final File baseOutputFile = createTempDir("testNoCoverageDueToFiltering");
+        final File output = IOUtils.createTempFileInDirectory( "testNoCoverageDueToFiltering", ".csv", baseOutputFile);
+
+        String cmd = "-R "+hg38Reference+" " +
+                "-I "+largeFileTestDir + "multiSampleSubsetted.bam " +
+                "-L "+ getTestFile("artificial.target_region.interval_list ")+
+                "--min-base-quality 5 --max-base-quality 4 -dels --print-base-counts -pt readgroup -pt sample -pt library -pt platform --output-format CSV";
```

Comment:
Is the idea here that because `--max-base-quality` is less than `--min-base-quality` you get no reads after filtering? Why does the tool even allow max to be less than min here?

Reply:
Good point, I suggest you bring it up with the gatk3 authors... This is a ported test.